tests/unit_tests/udfs/test_base_udf.py

```diff
@@ -0,0 +1,141 @@
+from typing import Union, Any, Tuple, List
+from unittest.mock import create_autospec, MagicMock, call, Mock
+
+import pytest
+from exasol_bucketfs_utils_python.bucketfs_factory import BucketFSFactory
+from exasol_udf_mock_python.column import Column
+from exasol_udf_mock_python.connection import Connection
+from exasol_udf_mock_python.mock_meta_data import MockMetaData
+
+from tests.unit_tests.utils_for_udf_tests import create_mock_exa_environment, create_mock_udf_context
+from tests.unit_tests.udfs.base_model_dummy_implementation import DummyImplementationUDF
+from exasol_transformers_extension.utils.huggingface_hub_bucketfs_model_transfer import ModelFactoryProtocol
+from tests.utils.mock_cast import mock_cast
+
+
+def create_mock_metadata() -> MockMetaData:
+    def udf_wrapper():
+        pass
+
+    meta = MockMetaData(
+        script_code_wrapper_function=udf_wrapper,
+        input_type="SET",
+        input_columns=[
+            Column("device_id", int, "INTEGER"),
+            Column("model_name", str, "VARCHAR(2000000)"),
+            Column("sub_dir", str, "VARCHAR(2000000)"),
+            Column("bucketfs_conn", str, "VARCHAR(2000000)"),
+            Column("token_conn", str, "VARCHAR(2000000)"),
+        ],
+        output_type="EMITS",
+        output_columns=[
+            Column("model_name", str, "VARCHAR(2000000)"),
+            Column("sub_dir", str, "VARCHAR(2000000)"),
+            Column("bucketfs_conn", str, "VARCHAR(2000000)"),
+            Column("token_conn", str, "VARCHAR(2000000)"),
+            Column("answer", bool, "BOOLEAN"),
+            Column("score", str, "DOUBLE"),
+            Column("error_message", str, "VARCHAR(2000000)")
+        ]
+    )
+    return meta
+
+
+@pytest.mark.parametrize("description, bucketfs_conn_name, bucketfs_conn ,"
+                         "sub_dir, model_name", [
```

Comment:
When I look at the tests, you probably, can also simplify them, because they share a lot of code, almost everything except input_data construction and asserts, so maybe extract that.

Reply:
or alternatively, I sometimes have only one test and a parameter which specifies, if the test is expected to produce an error, and have two different asserts depending on that parameter